Tests/Tests/AnalyticsBroadcasterSpec.swift

```diff
@@ -0,0 +1,132 @@
+final class AnalyticsTrackerMock: NSObject, Trackable {
+    var dictionary: [String: TrackerResult]?
+    private(set) var eventName: String?
+    private(set) var params: [String: Any]?
+
+    // MARK: - Trackable
+    func trackEvent(name: String, parameters: [String: Any]?) {
+        eventName = name
+        params = parameters
+        dictionary?[name] = TrackerResult(tracked: true, parameters: parameters)
+    }
+}
+
+// MARK: - SimpleContainerMock
+
+final class SimpleContainerMock: NSObject, SimpleDependenciesContainable {
+    public var tracker: Trackable?
+}
+
+/// This class tracks custom events.
+/// It creates event corressponding to each event, sends it to RAnalyticsManager's instance to process.
+final class RAnalyticsExternalCollector {
+
+    private let tracker: Trackable?
+
+    @available(*, unavailable)
+    init() {
+        self.tracker = nil
+    }
+
+    /// Creates an external collector
+    ///
+    /// - Parameters:
+    ///   - dependenciesContainer: The dependencies container.
+    ///         It requires intances of these types: Trackable
+    ///
+
+    /// - Returns: An instance of RAnalyticsExternalCollector.
+    init(dependenciesContainer: SimpleDependenciesContainable) {
+        self.tracker = dependenciesContainer.tracker
+
+        addCustomEventObserver()
+    }
+}
+
+// MARK: - Notification observers
+
+private extension RAnalyticsExternalCollector {
+    func addCustomEventObserver() {
+        let eventName = NSNotification.Name.sdkCustomEvent.rawValue
+        addNotificationName(eventName, selector: #selector(receiveCustomEventNotification(_:)))
+    }
+
+    @objc func receiveCustomEventNotification(_ notification: NSNotification) {
+        guard let parameters = notification.object as? [String: Any] else {
+            return
+        }
+        tracker?.trackEvent(name: NSNotification.Name.sdkCustomEvent.rawValue, parameters: parameters)
+    }
+
+    func addNotificationName(_ name: String, selector: Selector) {
+        NotificationCenter.default.addObserver(self,
+                                               selector: selector,
+                                               name: NSNotification.Name(rawValue: name),
+                                               object: nil)
+    }
+}
+
+// MARK: - AnalyticsBroadcasterSpec
+
+final class AnalyticsBroadcasterSpec: QuickSpec {
+    override func spec() {
+        describe("sendEventName") {
+            var dependenciesContainer: SimpleContainerMock!
+            beforeEach {
+                dependenciesContainer = SimpleContainerMock()
+                dependenciesContainer.tracker = AnalyticsTrackerMock()
+            }
+            it("should track sdkCustomEvent with eventName and eventData when an event is broadcasted with data") {
+                let externalCollector = RAnalyticsExternalCollector(dependenciesContainer: dependenciesContainer)
+                let tracker = (dependenciesContainer.tracker as? AnalyticsTrackerMock)
```

Comment:
Can the tests be more generic (and simpler)?
We just need to test that a notification can be observed which contains the expected name/data.

I don't think we need RAnalyticsExternalCollector, AnalyticsTracker, dependencies containers etc.